os/board/common/Kconfig

```diff
@@ -54,3 +54,53 @@ config FLASH_PART_NAME
 
 endif # FLASH_PARTITION
 endif # ARCH_USE_FLASH && (NFILE_DESCRIPTORS != 0)
+
+config ARCH_USE_EXTERNAL_FLASH
+	bool
+	default n
+	---help---
+		This shows that user enables External FLASH to use.
+
+if ARCH_USE_EXTERNAL_FLASH && (NFILE_DESCRIPTORS != 0)
+comment "External Flash Partition Options"
+
+config EXTERNAL_FLASH_PARTITION
+	bool "Enable partition support on external FLASH"
+	default n
+	select MTD
+	select MTD_PARTITION
+	---help---
+		Enables creation of partitions on the external FLASH
+
+if EXTERNAL_FLASH_PARTITION
+
+config EXTERNAL_FLASH_MINOR
+	int "Minor number for the FLASH device"
+	default 1
+	---help---
+		Sets the minor number for /dev node of the external flash
+		device.
+
+config EXTERNAL_FLASH_PART_SIZE
+	string "Flash partition size list (in KBytes)"
+	default "512,2048,"
+	---help---
+		Comma separated list of partition sizes in KB.
+
+config EXTERNAL_FLASH_PART_TYPE
+	string "Flash partition type list"
+	default "kernel,smartfs," if !APP_BINARY_SEPARATION
```

Comment:
I think kernel and smartfs are for APP_BINARY_SEPARATION. You'd better change with below.